core/tests/puppeteer-acceptance-tests/user-utilities/super-admin-utils.js

```diff
@@ -101,4 +101,444 @@ module.exports = class e2eSuperAdmin extends baseUser {
+        await _this.type(
+          '.e2e-test-commit-message-input', 'Create subtopic');
+        await _this.clickOn('.e2e-test-close-save-modal-button');
+      }
+    });
+    await this.page.waitForSelector(
+      '.e2e-test-save-topic-button:disabled');
+
+    // Assign diagnostic test skills.
+    // TODO (#19669): Once the issue is solved, remove this reload
+    // page statement and the saving topic changes statements just
+    // before the reload page statement.
+    await this.reloadPage();
+    await this.clickOn(
+      '.e2e-test-add-diagnostic-test-skill', { visible: true });
+    await this.select(
+      '.e2e-test-diagnostic-test-skill-selector',
+      await this.page.$eval(
+        '.e2e-test-diagnostic-test-skill-selector',
+        (dropdown, skillDescription) => {
+          for (const option of dropdown.options) {
+            if (option.text === skillDescription) {
+              return option.value;
+            }
+          }
+          return '';
+        },
+        diagnosticTestSkills[0].description),
+      { visible: true });
+
+    await this.clickOn('.e2e-test-save-topic-button:enabled');
+    await this.withinModal({
+      selector: 'oppia-topic-editor-save-modal',
+      whenOpened: async(_this) => {
+        await _this.type(
+          '.e2e-test-commit-message-input',
+          'Select diagnostic test skills');
+        await _this.clickOn('.e2e-test-close-save-modal-button');
+      }
+    });
+    await this.page.waitForSelector(
+      '.e2e-test-save-topic-button:disabled');
+
+    if (isPublished) {
+      // Navigates to the Topics and Skills Dashboard.
+      await Promise.all([
+        this.page.waitForNavigation(),
+        this.clickOn('.e2e-test-publish-topic-button')
+      ]);
+    }
+  }
+
+  async getTopicIdBy({ name }) {
+    await this.goto('http://localhost:8181/topics-and-skills-dashboard');
+    await this.clickOn('.e2e-test-topics-tab', { visible: true });
+
+    await Promise.all([
+      this.page.waitForNavigation({ waitUntil: 'networkidle0' }),
+      this.clickOn('.e2e-test-topic-name')
+    ]);
+
+    const topicEditorUrl = await this.page.url();
+
+    const topicIdMatcher = /\/\w+\#/;
+    const topicIdMatch = topicEditorUrl.match(topicIdMatcher)[0];
+    return topicIdMatch.substring(1, topicIdMatch.length - 1);
+  }
+
+  async createSkill({
+    description,
+    reviewMaterial,
+    misconception,
+    difficulties,
+    questionCount
+  }) {
+    await this.goto('http://localhost:8181/topics-and-skills-dashboard');
+
+    await this.clickOn('.e2e-test-create-skill-button', { visible: true });
+    await this.withinModal({
+      selector: 'oppia-create-new-skill-modal',
+      whenOpened: async(_this) => {
+        await _this.type(
+          '.e2e-test-new-skill-description-field', description);
+        await _this.clickOn('.e2e-test-open-concept-card');
+        await _this.type(
+          '.e2e-test-concept-card-text .e2e-test-rte', reviewMaterial,
```

Comment:
One general suggestion. Could you make these selectors named constants and put them at the top of the file? E.g.

```
SKILL_EXPLANATION_RTE = '.e2e-tset-concept-card-text .e2e-test-rte'
```

This is because we reuse some of these classes multiple times and also because I think it will be better for readability to have a human-recognizable name for what the element being interacted with actually is.

I am torn between whether SKILL_EXPLANATION_RTE Is enough or if we should name everything according to a convention like CSS_SELECTOR_SKILL_EXPLANATION_RTE or SKILL_EXPLANATION_RTE_SELECTOR -- on the one hand, the SELECTOR distinguishes it from other constants, but on the other, I'm wondering if the repeated CSS_SELECTOR or SELECTOR is going to make reading things more annoying. Another alternative is to put all the selectors in an object so that the syntax is `SELECTOR.SKILL_EXPLANATION_RTE`. Not sure which is best -- perhaps you can pick the approach which, in your view, makes the file as easily readable/grokkable as possible?

Reply:
The feature related to this review thread was removed.